protocol/communities/manager.go

```diff
@@ -5188,11 +5189,68 @@ func (m *Manager) promoteSelfToControlNode(community *Community, clock uint64) (
 		return false, err
 	}
 
+	if community.IsControlNode() {
+		err = m.handleCommunityEvents(community)
+		if err != nil {
+			return false, err
+		}
+	}
+
 	community.increaseClock()
 
 	return ownerChanged, nil
 }
 
+func (m *Manager) handleCommunityEvents(community *Community) error {
```

Comment:
It would be great to reuse this function in `HandleCommunityEventsMessage`, otherwise we'll have logic duplicated.

Reply:
I will be improving after manual testing that everything works properly